### PR TITLE
IGNITE-23958 Fixed security context propagation for async transactional operations

### PR DESCRIPTION
**Pull request by petrov-mg**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/testsuites/SecurityTestSuite.java [hunk 1]
```diff
@@ -143,6 +144,7 @@
     NodeSecurityContextPropagationTest.class,
     NodeJoinPermissionsTest.class,
     ActivationOnJoinWithoutPermissionsWithPersistenceTest.class,
+    SecurityContextInternalFuturePropagationTest.class
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/testsuites/SecurityTestSuite.java:147]: Please add comma at the EOL

Comment (petrov-mg) [reply modules/core/src/test/java/org/apache/ignite/testsuites/SecurityTestSuite.java]: Done.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/SecurityContextInternalFuturePropagationTest.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+            .setReadFromBackup(false)
+            .setBackups(1)
+            .setAffinity(new GridCacheModuloAffinityFunction(2, 1)));
+
+        awaitPartitionMapExchange();
+
+        try (IgniteClient cli = startClient("allowed_client")) {
+            for (int i = 0; i < PRELOADED_KEY_CNT; i++) {
+                cli.cache(DEFAULT_CACHE_NAME).put(i, i);
+            }
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/security/SecurityContextInternalFuturePropagationTest.java:215]: Redundant braces for one line statement

**Review by alex-plekhanov — APPROVED** [review]

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/SecurityContextInternalFuturePropagationTest.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+    /** */
+    @Parameterized.Parameters()
+    public static Iterable<Object[]> data() {
+        List<Object[]> res = new ArrayList<>();
+
+        for (Function<ClientCache<Object, Object>, IgniteClientFuture<?>> op : CACHE_OPERATIONS)
+            res.add(new Object[] {op});
+
+        return res;
```
[the remark this answers is not in the document]
Comment (petrov-mg) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/security/SecurityContextInternalFuturePropagationTest.java]: Done.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/SecurityContextInternalFuturePropagationTest.java (hunk 2 above)
Comment (petrov-mg) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/security/SecurityContextInternalFuturePropagationTest.java]: Done.

**Pull request merged**